src/aiida/repository/backend/disk_object_store.py

```diff
@@ -196,7 +198,7 @@ def maintain(  # type: ignore[override]
                 if not dry_run:
                     with get_progress_reporter()(total=1) as progress:
                         callback = create_callback(progress)
-                        container.pack_all_loose(compress=compress, callback=callback)
+                        container.pack_all_loose(compress=compress_mode, callback=callback)  # type: ignore[arg-type]
```

Comment:
This ignore needs to be solved in the disk-objecstore package, I'll open a PR there.